test/chain-test.js

```diff
@@ -766,6 +767,311 @@ describe('Chain', function() {
       'mandatory-script-verify-flag-failed');
   });
 
+  describe('Covenant limits', function() {
+    before(async () => {
+      await chain.reset(0);
+      wallet = new MemWallet({ network });
+
+      wallet.getNameStatus = async (nameHash) => {
+        assert(Buffer.isBuffer(nameHash));
+        const height = chain.height + 1;
+        const state = await chain.getNextState();
+        const hardened = state.hasHardening();
+        return chain.db.getNameStatus(nameHash, height, hardened);
+      };
+
+      // Because of the high volume of TXs sent by MemWallet in this test,
+      // we need to manually control the coins being used so we don't
+      // double-spend. MemWallet is too simple to track name covenants
+      // when they are unconfirmed, otherwise we could use wallet.addTX()
+      wallet.spendTX = (tx) => {
+        for (let i = 0; i < tx.inputs.length; i++) {
+          const input = tx.inputs[i];
+          const op = input.prevout.toKey();
+          wallet.removeCoin(op);
+        }
+      };
+    });
```

Comment:
Might be fixed by https://github.com/handshake-org/hsd/pull/629/commits/0820f67f602af8716eae00eeede3386f1c66b1d8